DNN Platform/DotNetNuke.Web/Api/ApiTokenAuthorizeAttribute.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Web.Api
+{
+    using DotNetNuke.Entities.Portals;
+    using DotNetNuke.Entities.Users;
+    using DotNetNuke.Web.Api.Auth.ApiTokens;
+    using DotNetNuke.Web.Api.Auth.ApiTokens.Models;
+
+    /// <summary>
+    /// Attribute to authorize apis based on the api token used. 
+    /// </summary>
+    public class ApiTokenAuthorizeAttribute : AuthorizeAttributeBase, IOverrideDefaultAuthLevel
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiTokenAuthorizeAttribute"/> class.
+        /// </summary>
+        /// <param name="key">The Key to authenticate api.</param>
+        /// <param name="resourceFile">The resource file for the token api.</param>
+        /// <param name="scope">The required api token scope.</param>
+        public ApiTokenAuthorizeAttribute(string key, string resourceFile, ApiTokenScope scope)
+        {
+            this.Key = key;
+            this.ResourceFile = resourceFile;
+            this.Scope = scope;
+        }
+
+        /// <summary>
+        /// Gets or sets the Key for the api token.
+        /// </summary>
+        public string Key { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the resource file for the api token.
+        /// </summary>
+        public string ResourceFile { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the required api token scope.
+        /// </summary>
+        public ApiTokenScope Scope { get; set; } = ApiTokenScope.User;
+
+        /// <summary>
+        /// Check if the request is authorized.
+        /// </summary>
+        /// <param name="context">The authentication filter context.</param>
+        /// <returns>True if authorized, false otherwise.</returns>
+        public override bool IsAuthorized(AuthFilterContext context)
+        {
+            var token = ApiTokenController.Instance.GetCurrentThreadApiToken();
+            if (token == null)
+            {
+                return false;
+            }
+
+            if (token.Scope != ApiTokenScope.Host)
+            {
+                var settings = ApiTokenSettings.GetSettings(PortalController.Instance.GetCurrentSettings().PortalId);
```

Comment:
Do we not have a kill-switch at the host level for this as well?

Reply:
Per discussion this would be done at the web.config level.  Are we ok with that difference in behavior?

Reply:
I'm okay with it, I don't think we need two ways to disable it.

Reply:
Works for me